src/Map.cpp

```diff
@@ -40,6 +41,7 @@ cMap::cMap(unsigned int a_ID, int a_CenterX, int a_CenterZ, cWorld * a_World, un
 	m_Scale(a_Scale),
 	m_CenterX(a_CenterX),
 	m_CenterZ(a_CenterZ),
+	m_Dirty(true),
```

Comment:
Is there a reason why this is true whereas the other initialisation on line 26 sets it to false? Doesn't make sense that adding a center position and scale would make the empty map dirty but not having them would be a clean map.

Reply:
m_Dirty means the map needs saving. Centre position and scale are things that are saved as part of the map files. Ergo, if you change them you need to save the map.

Reply:
What you meant to say was that this constructor is only called when creating a new map (which needs to be saved) while the other constructor is only used when loading a map (which does not need to be saved).